mysql/tests/test_statements.py

```diff
@@ -527,6 +529,62 @@ def run_query(q):
     assert metric['dd_commands'] == expected_metadata_payload['commands']
 
 
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+@pytest.mark.parametrize(
+    "set_reported_hostname,expected_reported_hostname",
+    [
+        (True, 'bob_hostname_override'),
+        (False, 'stubbed.hostname'),
+    ],
+)
+def test_statement_reported_hostname(
+    aggregator, dd_run_check, dbm_instance, datadog_agent, set_reported_hostname, expected_reported_hostname
+):
+    if set_reported_hostname:
+        dbm_instance['reported_hostname'] = expected_reported_hostname
+    mysql_check = MySql(common.CHECK_NAME, {}, [dbm_instance])
+
+    test_query = 'select * from information_schema.processlist where state in (\'starting\')'
```

Comment:
Does this test need to be so involved with the setup? To validate the `reported_hostname` setting I think all we need is to get a single valid payload of each data type and for that we don't need to manually run any queries. 

Just running the check twice should produce all the payloads we need purely from the traffic the agent generates on the database itself.